libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -274,7 +288,23 @@ void lua_scripts::reschedule_script(script_info *script) {
 }
 
 void lua_scripts::run(void) {
-    lua_State *L = luaL_newstate();
+    lua_State *L = nullptr;
+
+    // panic should be hooked first
+    if (setjmp(panic_jmp)) {
+        gcs().send_text(MAV_SEVERITY_INFO, "Lua: Caught a panic");
```

Comment:
is there any more diag stuff we could send such as which script caused it?

Reply:
There is, pushed it up at the actual panic handler.